connectd/peer_exchange_initmsg.c

```diff
@@ -206,6 +210,32 @@ struct io_plan *peer_exchange_initmsg(struct io_conn *conn,
 	tlvs->networks = tal_dup_arr(tlvs, struct bitcoin_blkid,
 				     &chainparams->genesis_blockhash, 1, 0);
 
+	/* BOLT-remote-address #1:
+	 * The sending node:
+	 * ...
+	 *  - SHOULD set `remote_addr` to reflect the remote IP address (and port) of an
+	 *    incoming connection, if the node is the receiver and the connection was done
+	 *    via IP.
+	 */
+	if (incoming && addr->itype == ADDR_INTERNAL_WIREADDR) {
+		if (addr->u.wireaddr.type == ADDR_TYPE_IPV4) {
+			remote_addr_v4 = tal(tlvs, struct remote_addr_ipv4);
+			remote_addr_v4->type = 1;
+			remote_addr_v4->port = addr->u.wireaddr.port;
+			memcpy(&remote_addr_v4->addr,
+			       addr->u.wireaddr.addr, 4);
+			tlvs->remote_addr = (u8*)remote_addr_v4;
+		}
+		if (addr->u.wireaddr.type == ADDR_TYPE_IPV6) {
+			remote_addr_v6 = tal(tlvs, struct remote_addr_ipv6);
+			remote_addr_v6->type = 2;
+			remote_addr_v6->port = addr->u.wireaddr.port;
+			memcpy(&remote_addr_v6->addr,
+			       addr->u.wireaddr.addr, 16);
+			tlvs->remote_addr = (u8*)remote_addr_v6;
+		}
+	}
```

Comment:
You need to check this is a public address, otherwise you're telling them about your internal network gateway.  See address_routable...

Reply:
I can add a `address_routeable()` check but allow localhost for testing. Actually a prior version of the PR had `address_routeable()` in it, but I removed it as part of the discussion in the RFC. Maybe we should add a line to the RFC that says "SHOULD NOT send non-public addresses for remote_addr".

Reply:
@rustyrussell See latest discussion on https://github.com/lightning/bolts/pull/917#issuecomment-1007311216

For a normal home NAT router this should not happen as a a node should see the public address, right?

Reply:
Update: I added `SHOULD NOT set private addresses as 'remote_addr'.` to the RFC and will add a check in the code but allow for localhost.